Userland/Libraries/LibDisassembly/riscv64/A.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "A.h"
+
+namespace Disassembly::RISCV64 {
+
+NonnullOwnPtr<InstructionImpl> parse_amo(u32 instruction)
+{
+    auto raw_parts = RawRType::parse(instruction);
+    auto is_acquire = (raw_parts.funct7 & 0b10) > 0;
+    auto is_release = (raw_parts.funct7 & 1) > 0;
+    auto width = MemoryAccessMode::from_funct3(raw_parts.funct3).width;
+
+    auto operation = AtomicMemoryOperation::Operation::Swap;
+    switch (raw_parts.funct7 >> 2) {
+    case 0b00010:
+    case 0b00011: {
+        auto is_sc = (raw_parts.funct7 & (1 << 2)) > 0;
+        return adopt_own(*new (nothrow) LoadReserveStoreConditional(is_sc ? LoadReserveStoreConditional::Operation::StoreConditional : LoadReserveStoreConditional::Operation::LoadReserve, is_acquire, is_release, width, raw_parts.rs1, raw_parts.rs2, raw_parts.rd));
+    }
+    case 0b00001:
```

Comment:
I am once again asking you to make this the enum values you are eventually/essentially assigning, so this becomes a cast-or-rejection pattern
(Seems like this did not make the cut for the the arithmetic instructions in G?)

Reply:
In this case it’s a good idea vs. the other ones where it didn’t actually match up with a single sub-instruction bitfield.